scheduler/feasible_test.go

```diff
@@ -416,64 +433,68 @@ func TestHostVolumeChecker_Sticky(t *testing.T) {
 			AttachmentMode: structs.CSIVolumeAttachmentModeFilesystem,
 		},
 	}
+	stickyJob := mock.Job()
+	stickyJob.TaskGroups[0].Volumes = stickyRequests
 
-	checker := NewHostVolumeChecker(ctx)
-
-	// alloc0 wants a previously registered volume ID that's available on node1
-	alloc0 := mock.Alloc()
-	alloc0.NodeID = nodes[1].ID
-	alloc0.HostVolumeIDs = []string{dhv.ID}
-
-	// alloc1 wants a volume ID that's available on node1 but hasn't used it
-	// before
-	alloc1 := mock.Alloc()
-	alloc1.NodeID = nodes[1].ID
+	existingClaims := []*structs.TaskGroupHostVolumeClaim{
+		{
+			Namespace:     structs.DefaultNamespace,
+			JobID:         stickyJob.ID,
+			TaskGroupName: stickyJob.TaskGroups[0].Name,
+			VolumeID:      dhv1.ID,
+			VolumeName:    dhv1.Name,
+		},
+	}
 
-	// alloc2 wants a volume ID that's unrelated
-	alloc2 := mock.Alloc()
-	alloc2.NodeID = nodes[1].ID
-	alloc2.HostVolumeIDs = []string{uuid.Generate()}
+	for _, claim := range existingClaims {
+		must.NoError(t, store.UpsertTaskGroupHostVolumeClaim(structs.MsgTypeTestSetup, 1000, claim))
+	}
 
-	// insert all the allocs into the state
-	must.NoError(t, store.UpsertAllocs(structs.MsgTypeTestSetup, 1000, []*structs.Allocation{alloc0, alloc1, alloc2}))
+	checker := NewHostVolumeChecker(ctx)
 
 	cases := []struct {
-		name   string
-		node   *structs.Node
-		alloc  *structs.Allocation
-		expect bool
+		name                             string
+		node                             *structs.Node
+		job                              *structs.Job
+		expect                           bool
+		expectedClaimsInTheVolumeChecker int
 	}{
 		{
-			"alloc asking for a sticky volume on an infeasible node",
+			"requesting a sticky volume on an infeasible node",
 			nodes[0],
-			alloc0,
+			stickyJob,
 			false,
+			1,
 		},
 		{
-			"alloc asking for a sticky volume on a feasible node",
+			"requesting a sticky volume on a feasible node, existing claim",
 			nodes[1],
-			alloc0,
+			stickyJob,
 			true,
+			0,
 		},
 		{
-			"alloc asking for a sticky volume on a feasible node for the first time",
+			"requesting a sticky volume on a feasible node, new claim",
 			nodes[1],
-			alloc1,
+			mock.Job(),
 			true,
+			0,
 		},
 		{
-			"alloc asking for an unrelated volume",
-			nodes[1],
-			alloc2,
+			"requesting a sticky volume on a feasible node, but there is an existing claim for another vol ID on a different node",
+			nodes[0],
+			stickyJob,
 			false,
+			1,
 		},
 	}
 
 	for _, tc := range cases {
 		t.Run(tc.name, func(t *testing.T) {
-			checker.SetVolumes(tc.alloc.Name, structs.DefaultNamespace, stickyRequest, tc.alloc.HostVolumeIDs)
+			checker.SetVolumes(mock.Alloc().Name, structs.DefaultNamespace, tc.job.ID, tc.job.TaskGroups[0].Name, stickyRequests)
```

Comment:
If I'm understanding correctly, `SetVolumes` doesn't clear the `claim` field, it appends to it, which makes the tests ordered. It seems like that's fine now because of the specific test cases we have, but maybe to make life easier on future test writers we should move `NewHostVolumeChecker` call into the subtest?